internal/datastore/common/sql_test.go

```diff
@@ -686,28 +596,263 @@ func TestSchemaQueryFilterer(t *testing.T) {
+			},
+			options: []options.QueryOptionsOption{
+				options.WithSkipCaveats(true),
+			},
+			expectedSkipCaveats: true,
+			expectedSQL:         "SELECT object_id, relation, subject_ns, subject_object_id, subject_relation, expiration FROM relationtuples WHERE (expiration IS NULL OR expiration > NOW()) AND ns = ?",
+			expectedArgs:        []any{"sometype"},
+		},
+		{
+			name: "filter by just subject type",
+			run: func(filterer SchemaQueryFilterer) SchemaQueryFilterer {
+				return filterer.FilterToSubjectFilter(&v1.SubjectFilter{
+					SubjectType: "subns",
+				})
+			},
+			expectedSQL:  "SELECT ns, object_id, relation, subject_object_id, subject_relation, caveat, caveat_context, expiration FROM relationtuples WHERE (expiration IS NULL OR expiration > NOW()) AND subject_ns = ?",
+			expectedArgs: []any{"subns"},
+		},
+		{
+			name: "filter by just subject type and subject ID",
+			run: func(filterer SchemaQueryFilterer) SchemaQueryFilterer {
+				return filterer.FilterToSubjectFilter(&v1.SubjectFilter{
+					SubjectType:       "subns",
+					OptionalSubjectId: "subid",
+				})
+			},
+			expectedSQL:  "SELECT ns, object_id, relation, subject_relation, caveat, caveat_context, expiration FROM relationtuples WHERE (expiration IS NULL OR expiration > NOW()) AND subject_ns = ? AND subject_object_id = ?",
+			expectedArgs: []any{"subns", "subid"},
+		},
+		{
+			name: "filter by just subject type and subject relation",
+			run: func(filterer SchemaQueryFilterer) SchemaQueryFilterer {
+				return filterer.FilterToSubjectFilter(&v1.SubjectFilter{
+					SubjectType: "subns",
+					OptionalRelation: &v1.SubjectFilter_RelationFilter{
+						Relation: "subrel",
+					},
+				})
+			},
+			expectedSQL:  "SELECT ns, object_id, relation, subject_object_id, caveat, caveat_context, expiration FROM relationtuples WHERE (expiration IS NULL OR expiration > NOW()) AND subject_ns = ? AND subject_relation = ?",
+			expectedArgs: []any{"subns", "subrel"},
+		},
+		{
+			name: "filter by just subject type and subject ID and relation",
+			run: func(filterer SchemaQueryFilterer) SchemaQueryFilterer {
+				return filterer.FilterToSubjectFilter(&v1.SubjectFilter{
+					SubjectType:       "subns",
+					OptionalSubjectId: "subid",
+					OptionalRelation: &v1.SubjectFilter_RelationFilter{
+						Relation: "subrel",
+					},
+				})
+			},
+			expectedSQL:  "SELECT ns, object_id, relation, caveat, caveat_context, expiration FROM relationtuples WHERE (expiration IS NULL OR expiration > NOW()) AND subject_ns = ? AND subject_object_id = ? AND subject_relation = ?",
+			expectedArgs: []any{"subns", "subid", "subrel"},
+		},
+		{
+			name: "filter by multiple subject types, but static subject ID",
+			run: func(filterer SchemaQueryFilterer) SchemaQueryFilterer {
+				return filterer.FilterToSubjectFilter(&v1.SubjectFilter{
+					SubjectType:       "subns",
+					OptionalSubjectId: "subid",
+				}).FilterToSubjectFilter(&v1.SubjectFilter{
```

Comment:
where would this query be used? same id but in two namespaces?

Reply:
Likely no where, but I wanted to exercise an odd corner case to make sure the logic was sound